ttnn/cpp/ttnn/operations/moreh/moreh_fold/device/fold_program_factory_rm.cpp

```diff
@@ -0,0 +1,217 @@
+        program, reader_kernel_file, all_cores, reader_compile_time_args, reader_defines);
+    const auto writer_kernel_id = tt::operations::primary::CreateWriteKernel(
+        program, writer_kernel_file, all_cores, writer_compile_time_args, writer_defines);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      RuntimeArgs SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    const auto input_addr = input.buffer()->address();
+    const auto output_addr = output.buffer()->address();
+
+    uint32_t start_id = 0;
+    auto cores = grid_to_cores(num_cores, num_cores_x, num_cores_y, false);
+    uint32_t g1_numcores = core_group_1.num_cores();
+    for (uint32_t i = 0; i < cores.size(); ++i) {
+        const CoreCoord& core = cores.at(i);
+        uint32_t num_units_per_core = i < g1_numcores ? num_units_per_core_group_1 : num_units_per_core_group_2;
+
+        vector<uint32_t> reader_args = {
+            input.buffer()->address(),
+            N,
+            C,
+            H,
+            W,
+            kernel_size_h,
+            kernel_size_w,
+            stride_h,
+            stride_w,
+            padding_h,
+            padding_w,
+            dilation_h,
+            dilation_w,
+            LH,
+            LW,
+            aligned_input_cb_page_size,
+            aligned_output_cb_page_size,
+            start_id,
+            num_units_per_core};
+
+        vector<uint32_t> writer_args = {
+            output.buffer()->address(), aligned_output_cb_page_size, start_id, num_units_per_core};
+
+        SetRuntimeArgs(program, reader_kernel_id, core, reader_args);
+        SetRuntimeArgs(program, writer_kernel_id, core, writer_args);
+        start_id += num_units_per_core;
+    }
+
+    return {std::move(program), {reader_kernel_id, writer_kernel_id, cores}};
+}
+
+void MorehFoldOperation::ProgramFactory::override_runtime_arguments(
+    cached_program_t& cached_program,
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& tensor_return_value) {
+    auto& program = cached_program.program;
+    auto& reader_kernel_id = cached_program.shared_variables.unary_reader_kernel_id;
+    auto& writer_kernel_id = cached_program.shared_variables.unary_writer_kernel_id;
+    auto& cores = cached_program.shared_variables.cores;
+
+    auto input_buffer = tensor_args.input.buffer();
+    auto output_buffer = tensor_return_value.buffer();
+
+    for (const auto& core : cores) {
+        {
+            auto& runtime_args = GetRuntimeArgs(program, reader_kernel_id, core);
+            runtime_args[0] = input_buffer->address();
+        }
+
+        {
+            auto& runtime_args = GetRuntimeArgs(program, writer_kernel_id, core);
+            runtime_args[0] = output_buffer->address();
+        }
+    }
+}
+}  // namespace ttnn::operations::moreh::moreh_fold
```

Comment:
Should be input_buffer_address = tensor_args.input.buffer()->address()
Then use it in the loop.

Reply:
Resolved in last commit.